MainWindow.xaml.cs

```diff
@@ -133,16 +140,47 @@ private void txt_OutputDirectory_TextChanged(object sender, TextChangedEventArgs
 
         #region Helpers
 
+        /// <summary>
+        /// Populate disc type according to system type
+        private void PopulateDiscTypeAccordingToChosenSystem()
+        {
+          cmb_DiscType.SelectedIndex = -1;
+
+          var currentSystem = cmb_SystemType.SelectedItem as Tuple<string, KnownSystem?>;
+
+          if (currentSystem != null && currentSystem.Item2 != KnownSystem.NONE)
+          {
+
+            List<Tuple<string, DiscType?>> allowedDiscTypesForSystem = Utilities.Validation.GetValidDiscTypes(currentSystem.Item2)
```

Comment:
Why not just modify what comes out of GetValidDiscTypes? It seems that would make the code cleaner.